core/src/main/java/org/kiwix/kiwixmobile/core/settings/SliderPreference.java

```diff
@@ -168,12 +174,22 @@ protected void onBindView(View view) {
   private void setGrayState(View view) {
     SharedPreferenceUtil sharedPreferenceUtil = new SharedPreferenceUtil(getContext());
     boolean enabled = sharedPreferenceUtil.getPrefZoomEnabled();
+    NightModeConfig nightMode = new NightModeConfig(sharedPreferenceUtil,getContext());
+    boolean nightMode_enabled = nightMode.isNightModeActive();
 
     TextView titleView = view.findViewById(android.R.id.title);
     TextView summaryTV = view.findViewById(android.R.id.summary);
     if (!enabled) {
       titleView.setTextColor(Color.GRAY);
       summaryTV.setTextColor(Color.GRAY);
+    }else{
+      if(nightMode_enabled){
+        titleView.setTextColor(Color.WHITE);
+        summaryTV.setTextColor(Color.WHITE);
+      }else{
+        titleView.setTextColor(Color.BLACK);
+        summaryTV.setTextColor(Color.BLACK);
```

Comment:
This whole logic can have the scope reduced, what actually happens here is you decide the textColor and then repeat the text setting code. 
Something like
```
int color = resolveColorFromState()
titleView.setTextColor(color);
summaryTV.setTextColor(color);

private int resolveColorFromState(){
  if(isEnabled){
    return nightModeConfig.isNightModeActive ? WHITE : BLACK
  }
  return GRAY;
}
```
and that is just a rough outline, AS will probably offer to reduce that even further